processing/src/main/java/org/apache/carbondata/processing/loading/sort/impl/UnsafeParallelReadMergeSorterWithColumnRangeImpl.java

```diff
@@ -99,6 +101,8 @@ public void initialize(SortParameters sortParameters) {
     UnsafeSortDataRows[] sortDataRows = new UnsafeSortDataRows[columnRangeInfo.getNumOfRanges()];
     intermediateFileMergers = new UnsafeIntermediateMerger[columnRangeInfo.getNumOfRanges()];
     SortParameters[] sortParameterArray = new SortParameters[columnRangeInfo.getNumOfRanges()];
+    this.writeService = Executors.newFixedThreadPool(originSortParameters.getNumberOfCores(),
```

Comment:
If you increase `carbon.number.of.cores.while.loading`, there will be more UnsafeSortDataRows and writing temp files can finish faster without any of these changes.

Is it necessary to introduce another multi-thread here ?
please tell your opinion @kevinjmh @kumarvishal09

Reply:
@ajantha-bhat Good point. So the only difference is adding threads horizontally or vertically.  If each thread takes same time to process the data and writes at same time, performance may degrade caused by IO preemption. But the different may not big when number of input split is large enough. @shunlean could you please do some test to confirm ?

Reply:
@kevinjmh : Yes, If cores are available adding threads horizontally can speedup not just sort, but other steps in data loading also.
If cores are not available, adding threads vertically also no use as they will end up waiting for cpu. 

so, I felt. This PR changes not required and user can increase `carbon.number.of.cores.while.loading`